app/types/signupForm.ts

```diff
@@ -0,0 +1,9 @@
+export type ValidationFunction = (value: string) => string | null | undefined;
```

Comment:
Our validation functions all seem to return either a string or null. Is there a scenario where they would return undefined? I think this is a good area to be very explicit about what we expect to receive.